io.openems.edge.io.shelly/src/io/openems/edge/io/shelly/shelly25/IoShelly25Impl.java

```diff
@@ -100,68 +113,84 @@ public void handleEvent(Event event) {
 		}
 
 		switch (event.getTopic()) {
-		case EdgeEventConstants.TOPIC_CYCLE_BEFORE_PROCESS_IMAGE:
-			this.eventBeforeProcessImage();
-			break;
-
-		case EdgeEventConstants.TOPIC_CYCLE_EXECUTE_WRITE:
-			this.eventExecuteWrite();
-			break;
+		case EdgeEventConstants.TOPIC_CYCLE_EXECUTE_WRITE //
+			-> this.eventExecuteWrite();
 		}
 	}
 
 	/**
 	 * Execute on Cycle Event "Before Process Image".
+	 * 
+	 * @param result The JSON element containing the result of the HTTP request.
+	 * @param error  The throwable error, if any occurred during the HTTP request.
 	 */
-	private void eventBeforeProcessImage() {
-		Boolean relay1IsOn;
-		Boolean relay2IsOn;
-		try {
-			var json = this.shellyApi.getStatus();
-			var relays = JsonUtils.getAsJsonArray(json, "relays");
-			var relay1 = JsonUtils.getAsJsonObject(relays.get(0));
-			relay1IsOn = JsonUtils.getAsBoolean(relay1, "ison");
-			var relay2 = JsonUtils.getAsJsonObject(relays.get(1));
-			relay2IsOn = JsonUtils.getAsBoolean(relay2, "ison");
+	private void processHttpResult(JsonElement result, Throwable error) {
+		this._setSlaveCommunicationFailed(result == null);
+		Boolean relay1 = null;
+		Boolean overtemp1 = null;
+		Boolean overpower1 = null;
+		Boolean relay2 = null;
+		Boolean overtemp2 = null;
+		Boolean overpower2 = null;
 
+		try {
+			JsonElement jsonElement = JsonUtils.getAsJsonElement(result);
+			final var relays = JsonUtils.getAsJsonArray(jsonElement, "relays");
+			for (int i = 0; i < relays.size(); i++) {
+				final var relay = JsonUtils.getAsJsonObject(relays.get(i));
+				final var relayIson = JsonUtils.getAsBoolean(relay, "ison");
+				final var relayOverpower = JsonUtils.getAsBoolean(relay, "overpower");
+				final var relayOvertemp = JsonUtils.getAsBoolean(relay, "overtemperature");
+				this.digitalOutputChannels[i].setNextWriteValue(relayIson);
```

Comment:
Its wrong to set the write value here. The write value is only for setting a value which then should be written to the external device. In the "read" method you only want to set the value you re reading which you re already doing with `this._setRelay1(relay1IsOn)`. Most of the times the write value is only set by other components which are using the relay to trigger something and not the component which declares it.